iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/api/RunInstruction.java

```diff
@@ -28,5 +28,14 @@
  * </pre>
  */
 public interface RunInstruction extends CommandInstruction {
+
+  /**
+   * @return the options as a list of {@link Flag}s.
+   */
   List<Flag> options();
+
+  /**
+   * @return true if the arguments of this RunInstruction are in {@link HereDocument} format.
+   */
+  boolean containsHeredoc();
```

Comment:
Maybe introduce a enum `HasArguments.Kind` and return kind here instead? If we need to distinguish exec form and shell form, we would need to introduce two more boolean methods.